lib/include/public/ISemanticContext.hpp

```diff
@@ -109,12 +109,39 @@ namespace MAT_NS_BEGIN
         /// <param name="deviceClass">Device class.</param>
         DECLARE_COMMONFIELD(DeviceClass, COMMONFIELDS_DEVICE_CLASS);
 
-          /// <summary>
+        /// <summary>
         /// Set the device orgId context information of telemetry event.
         /// </summary>
         /// <param name="deviceClass">Device orgId</param>
         DECLARE_COMMONFIELD(DeviceOrgId, COMMONFIELDS_DEVICE_ORGID);
 
+        /// <summary>
+        /// Set the W3C TraceContext TraceId information of telemetry event.
+        /// </summary>
+        /// <param name="traceId">TraceContext TraceId</param>
+        DECLARE_COMMONFIELD(TraceId, COMMONFIELDS_DT_TRACEID);
+
+        /// <summary>
+        /// Set the W3C TraceContext SpanId information of telemetry event.
+        /// </summary>
+        /// <param name="spanId">TraceContext SpanId</param>
+        DECLARE_COMMONFIELD(SpanId, COMMONFIELDS_DT_SPANID);
+
+        /// <summary>
+        /// Set the W3C TraceContext TraceFlags information of telemetry event.
+        /// </summary>
+        /// <param name="TraceFlags">TraceContext TraceFlags</param>
+        virtual void SetTraceFlags(uint8_t traceFlags)
```

Comment:
nit: the current spec only supports one flag (`sampled`) so I wonder if we should rather:

- add a `SetSampled(bool)` method, or
- change this to take a bitfield, or
- provide an enum or a TRACEFLAGS_SAMPLED constant to use here instead of '1'

(the last is probably best since it doesn't constrain the caller)

Reply:
W3C spec and Common Schema support `traceFlags`:

<img width="586" alt="image" src="https://user-images.githubusercontent.com/34072974/199781687-91e63c7b-e61c-4c4b-89dd-0efc7382acb3.png">

An integer representation of the [W3C TraceContext trace-flags](https://www.w3.org/TR/trace-context/#trace-flags).

Reply:
Yes I am saying that we could have
```cpp
struct TraceFlags {
  unsigned char sampled: 1,
  unsigned char reserved: 7
};
...
virtual void SetTraceFlags(TraceFlags traceFlags)
...
TraceFlags flags;
flags.sampled = 1;
context.SetTraceFlags(flags);
```
or rather
```cpp
enum TraceFlags {
  Sampled: 1,
};
...
context.SetTraceFlags(TraceFlags::Sampled);
``` 
rather than passing a raw uint8_t.

Reply:
We could later on add any additional method overloads we want. Indeed, we could add yet another method what you're proposing. I would rather go with an additional method that just does `SetTraceContext`. My goal was to expose the bare min set of flags, as per spec:

- in Common Schema spec this is `int`
- in W3C TraceContext spec this described as `byte` / `uint8_t`

Thus, I followed the W3C TraceContext guidelines in this regard. You can see that other implementations, e.g. [Elastic Node Traceparent package](https://github.com/elastic/node-traceparent#traceparentflags) - did not elaborate into what flags could mean exactly. It's just flags number, no more no less than what the transport layer is expecting to get.

We can add more decoration later. It's add-ons, not the necessity.

Reply:
Ok sounds good not to over-specify.